lib/std/once.nim

```diff
@@ -0,0 +1,31 @@
+import atomics
```

Comment:
this doesn't work:
```nim
when true:
  import std/once
  var block1: Once
  initOnce(block1)
  var thr: array[0..4, Thread[void]]
  proc threadFunc() {.thread.} =
    var count = 0
    for i in 1 .. 10:
      once(block1):
        echo "in once"
  for i in 0..high(thr):
    createThread(thr[i], threadFunc)
  joinThreads(thr)
```

depending on the runs, you'll get anywhere from 1 to 4 `in once`

Reply:
fixed it

Reply:
```nim
block:
  var thr: array[0..4, Thread[void]]
  proc threadFunc() {.thread.} =
    var block1: Once
    initOnce(block1)
    for i in 1 .. 10:
      once(block1):
        echo "in once"
  for i in 0..high(thr):
    createThread(thr[i], threadFunc)
  joinThreads(thr)
```
This also works for me: print `in once` five times.